server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java

```diff
@@ -1111,7 +1207,8 @@ public void cleanUpInternalState() {
 
     // This method does the following:
     //
-    // 1. Removes entries from RUNNING_CACHE that are not really running
+    // 1. Removes entries from RUNNING_CACHE and TIME_ORDERED_RUNNING_SET that are not really
```

Comment:
Can not assume that everything in TIME_ORDERED_RUNNING_SET is also in RUNNING_CACHE.  So a few lines down it need to take the union of the sets.

```
final Set<ExternalCompactionId> idsSnapshot = Set.copyOf(Sets.union(RUNNING_CACHE.keySet(), TIME_ORDERED_RUNNING_SET));
```